src/components/Dialog/DialogsManager.ts

```diff
@@ -0,0 +1,133 @@
+import { StateStore } from 'stream-chat';
+
+type DialogId = string;
+
+export type GetOrCreateParams = {
+  id: DialogId;
+};
+
+export type Dialog = {
+  close: () => void;
+  id: DialogId;
+  isOpen: boolean | undefined;
+  open: (zIndex?: number) => void;
+  remove: () => void;
+  toggle: () => void;
+  toggleSingle: () => void;
+};
+
+type DialogInitOptions = {
+  id?: string;
+};
+
+type Dialogs = Record<DialogId, Dialog>;
+
+type DialogsManagerState = {
+  dialogs: Dialogs;
+  openDialogCount: number;
+};
+
+export class DialogsManager {
+  id: string;
+  state = new StateStore<DialogsManagerState>({
+    dialogs: {},
+    openDialogCount: 0,
+  });
+
+  constructor({ id }: DialogInitOptions = {}) {
+    this.id = id ?? new Date().getTime().toString();
+  }
+
+  getOrCreate({ id }: GetOrCreateParams) {
+    let dialog = this.state.getLatestValue().dialogs[id];
+    if (!dialog) {
+      dialog = {
+        close: () => {
+          this.close(id);
+        },
+        id,
+        isOpen: false,
+        open: () => {
+          this.open({ id });
+        },
+        remove: () => {
+          this.remove(id);
+        },
+        toggle: () => {
+          this.toggleOpen({ id });
+        },
+        toggleSingle: () => {
+          this.toggleOpenSingle({ id });
+        },
+      };
+      this.state.next((current) => ({
+        ...current,
+        ...{ dialogs: { ...current.dialogs, [id]: dialog } },
+      }));
+    }
+    return dialog;
+  }
+
+  open(params: GetOrCreateParams, single?: boolean) {
```

Comment:
Nit: Can we merge options so there's only one parameter?

Rename of the actual property, `single` doesn't feel right; `closeRest` or `closeOpened` might better describe what's going to happen if it's set to `true`.
```suggestion
  open(params: GetOrCreateParams, closeOpened?: boolean) {
```

Reply:
The `params` arg was intended as a seed of dialog data. The data was gradually reduced to `id`, however I decided to keep it as object in case we would like to add more dialog props in the future. The second param is not related to the dialog being open but is a config param on how to handle opening. That is why I did not want to merge it into `GetOrGreateParams`.

Reply:
I will rename `single` to `closeRest`.

Reply:
done